### PR TITLE
CI : run skydive tests after building/deploying vagrant

### PR DESCRIPTION
**Pull request by nplanel**

### CODE REVIEW

**Comment by nplanel** [comment]
run skydive-functional-tests-backend-elasticsearch

**Comment by nplanel** [comment]
run skydive-vagrant-tests

**Comment by nplanel** [comment]
run skydive-functional-tests-backend-elasticsearch

**Comment by nplanel** [comment]
run create-vagrant-boxes

**Comment by nplanel** [comment]
run tests

**Comment by nplanel** [comment]
run vagrant-tests

**Comment by lebauce** [comment]
run skydive-unit-tests

**Comment by lebauce** [comment]
run skydive-create-vagrant-boxes

**Comment by nplanel** [comment]
run tests

**Comment by nplanel** [comment]
run skydive-create-vagrant-boxes

**Comment by nplanel** [comment]
run skydive-vagrant-tests

**Comment by lebauce** [comment]
The permission denied comes from the fact we change the SSH public key during provisoning. You should probably only add the key:

```
diff --git a/contrib/dev/Vagrantfile b/contrib/dev/Vagrantfile
index 54fac8a4..8550847d 100644
--- a/contrib/dev/Vagrantfile
+++ b/contrib/dev/Vagrantfile
@@ -24,7 +24,7 @@ sudo akmods --kernels $latest
 SCRIPT
 
 $vagrant_insecure_key_script = <<SCRIPT
-echo "ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEA6NF8iallvQVp22WDkTkyrtvp9eWW6A8YVr+kz4TjGYe7gHzIw+niNltGEFHzD8+v1I2YJ6oXevct1YeS0o9HZyN1Q9qgCgzUFtdOKLv6IedplqoPkcmF0aYet2PkEDo3MlTBckFXPITAMzF8dJSIFo9D8HfdOV0IAdx4O7PtixWKn5y2hMNG0zQPyUecp4pzC6kivAIhyfHilFR61RGL+GPXQ2MWZWFYbAGjyiYJnAmCP3NOTd0jMZEnDkbUvxhMmBYSdETk1rRgm+R4LOzFUGaHqHDLKLX+FIPKcF96hrucXzcWyLbIbEgE98OHlnVYCzRdK8jlqm8tehUc9c9WhQ== vagrant insecure public key" > ~/.ssh/authorized_keys
+echo "ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEA6NF8iallvQVp22WDkTkyrtvp9eWW6A8YVr+kz4TjGYe7gHzIw+niNltGEFHzD8+v1I2YJ6oXevct1YeS0o9HZyN1Q9qgCgzUFtdOKLv6IedplqoPkcmF0aYet2PkEDo3MlTBckFXPITAMzF8dJSIFo9D8HfdOV0IAdx4O7PtixWKn5y2hMNG0zQPyUecp4pzC6kivAIhyfHilFR61RGL+GPXQ2MWZWFYbAGjyiYJnAmCP3NOTd0jMZEnDkbUvxhMmBYSdETk1rRgm+R4LOzFUGaHqHDLKLX+FIPKcF96hrucXzcWyLbIbEgE98OHlnVYCzRdK8jlqm8tehUc9c9WhQ== vagrant insecure public key" >> ~/.ssh/authorized_keys
 SCRIPT
 $write_motd_script = <<SCRIPT
 ```

File: scripts/ci/create-vagrant-boxes.sh [hunk 1]
```diff
@@ -25,6 +25,11 @@ do
     PREPARE_BOX=true vagrant up --provider=$provider
     [ "$provider" = "libvirt" ] && sudo chmod a+r /var/lib/libvirt/images/dev_dev.img || true
 
+    # skydive testing
+    vagrant ssh -c 'set -e; src/github.com/skydive-project/skydive; make test functional; curl -XDELETE "localhost:9200/skydive*"'
```
Comment (lebauce) [inline scripts/ci/create-vagrant-boxes.sh]: You probably mean "cd src/github.com/skydive-project/skydive"

File: scripts/ci/run-vagrant-tests.sh [hunk 2]
```diff
@@ -20,6 +20,15 @@ do
   DEPLOYMENT_MODE=$mode vagrant up --provision-with common
   DEPLOYMENT_MODE=$mode vagrant provision
   retcode=$?
+
+  # skydive testing
+  if [ $retcode -ne 0 ] ; then
```
Comment (lebauce) [inline scripts/ci/run-vagrant-tests.sh]: In this case, we probably want to compile the tests on the host, copy them to the guest and run them. This is done in https://github.com/skydive-project/skydive/blob/master/scripts/ci/run-kolla-tests.sh#L9 and https://github.com/skydive-project/skydive/blob/master/scripts/ci/kolla/Vagrantfile#L14

Comment (nplanel) [reply scripts/ci/run-vagrant-tests.sh]: I don't like so much not using the environment / vm that we designed to run for.
I mean we can do the trick to compile on the host and push to the vm, but in this case why working hard to setup the vm correctly with all the dev packages ready to build and run ?

Meanwhile I like the idea of Vagrantfile integrate the test script, that we can trig/run by setting an environment variable

**Comment by lebauce** [comment]
run skydive-create-vagrant-boxes

**Comment by lebauce** [comment]
run skydive-vagrant-tests

**Comment by nplanel** [comment]
run skydive-create-vagrant-boxes

**Comment by nplanel** [comment]
run skydive-vagrant-tests

**Comment by nplanel** [comment]
run skydive-vagrant-tests

**Comment by nplanel** [comment]
run skydive-create-vagrant-boxes

**Comment by nplanel** [comment]
run skydive-vagrant-tests

**Comment by nplanel** [comment]
run skydive-create-vagrant-boxes

**Comment by nplanel** [comment]
run skydive-vagrant-tests

**Comment by nplanel** [comment]
run skydive-create-vagrant-boxes

**Comment by nplanel** [comment]
run skydive-unit-tests

**Comment by nplanel** [comment]
run skydive-unit-tests

**Comment by nplanel** [comment]
run skydive-create-vagrant-boxes

**Comment by nplanel** [comment]
run skydive-vagrant-tests

**Comment by nplanel** [comment]
run skydive-create-vagrant-boxes

File: contrib/dev/Vagrantfile [hunk 3]
```diff
@@ -173,7 +173,8 @@ Vagrant.configure(2) do |config|
       dev.vm.provision :ansible_local do |ansible|
         ansible_extra_vars = {
           "skydive_extra_config" => $skydive_extra_config,
-          "skydive_fabric_default_interface" => "eth0"
+          "skydive_fabric_default_interface" => "eth0",
+          "skydive_release" => ENV.fetch("SKYDIVE_RELEASE", "v0.17.0"),
```
Comment (lebauce) [inline contrib/dev/Vagrantfile]: There is already a default value for the skydive_release  in the ansible roles. I would prefer to specify `skydive_release` only if the environment variable is defined

File: contrib/vagrant/Vagrantfile [hunk 4]
```diff
@@ -149,7 +149,8 @@ Vagrant.configure(2) do |config|
             "skydive_extra_config" => $skydive_extra_config,
             "skydive_nic" => "ansible_eth1",
             "skydive_fabric_default_interface" => "eth1",
-            "skydive_listen_ip" => "0.0.0.0"
+            "skydive_listen_ip" => "0.0.0.0",
+            "skydive_release" => ENV.fetch("SKYDIVE_RELEASE", "v0.17.0"),
```
Comment (lebauce) [inline contrib/vagrant/Vagrantfile]: Same as above

File: scripts/ci/create-vagrant-boxes.sh [hunk 5]
```diff
@@ -11,7 +11,10 @@ fi
 
 dir="$(dirname "$0")"
 
-VERSION="$(grep 'skydive_release:' ${dir}/../../contrib/ansible/roles/skydive_common/defaults/main.yml | cut -f 2 -d ' ' | tr -d 'v')"
+
+TAG=`echo $REF | awk -F '/' '{print $NF}'`
```
Comment (lebauce) [inline scripts/ci/create-vagrant-boxes.sh]: When the job is triggered, REF looks like `refs/merge/1025`

**Comment by nplanel** [comment]
run create-vagrant-boxes

**Comment by nplanel** [comment]
run create-vagrant-boxes

**Comment by nplanel** [comment]
run skydive-vagrant-tests

**Comment by nplanel** [comment]
run create-vagrant-boxes run skydive-vagrant-tests

**Comment by nplanel** [comment]
run create-vagrant-boxes run skydive-vagrant-tests

**Comment by nplanel** [comment]
run create-vagrant-boxes

**Comment by nplanel** [comment]
run create-vagrant-boxes

**Comment by nplanel** [comment]
run create-vagrant-boxes

**Comment by nplanel** [comment]
run create-vagrant-boxes run skydive-vagrant-tests

**Comment by nplanel** [comment]
run create-vagrant-boxes run skydive-vagrant-tests

**Comment by nplanel** [comment]
run create-vagrant-boxes

**Comment by nplanel** [comment]
run create-vagrant-boxes run skydive-vagrant-tests

**Comment by nplanel** [comment]
run create-vagrant-boxes run skydive-vagrant-tests

**Comment by nplanel** [comment]
run create-vagrant-boxes run skydive-vagrant-tests

**Comment by nplanel** [comment]
run create-vagrant-boxes run skydive-vagrant-tests

**Comment by nplanel** [comment]
run create-vagrant-boxes

**Comment by nplanel** [comment]
run create-vagrant-boxes

**Comment by nplanel** [comment]
run create-vagrant-boxes

**Comment by nplanel** [comment]
run create-vagrant-boxes run skydive-vagrant-tests

**Comment by nplanel** [comment]
run run create-vagrant-boxes

**Comment by nplanel** [comment]
run skydive-vagrant-tests

**Comment by nplanel** [comment]
run skydive-vagrant-tests

**Comment by nplanel** [comment]
run skydive-vagrant-tests

**Comment by nplanel** [comment]
run skydive-vagrant-tests

**Comment by nplanel** [comment]
run skydive-create-vagrant-boxes Pending — Build started for merge commit.

**Comment by nplanel** [comment]
run skydive-create-vagrant-boxes

**Comment by nplanel** [comment]
run skydive-create-vagrant-boxes

**Comment by nplanel** [comment]
run skydive-create-vagrant-boxes

**Comment by nplanel** [comment]
run skydive-create-vagrant-boxes

**Comment by nplanel** [comment]
run skydive-create-vagrant-boxes

**Comment by nplanel** [comment]
run skydive-go-fmt

**Comment by nplanel** [comment]
run skydive-go-fmt Pending — Build started for merge commit.

**Comment by nplanel** [comment]
fdsdsfs

**Comment by nplanel** [comment]
run create-vagrant-boxes run skydive-vagrant-tests

**Comment by nplanel** [comment]
ok

**Comment by nplanel** [comment]
ok

**Comment by nplanel** [comment]
ok

**Comment by nplanel** [comment]
run skydive-create-vagrant-boxes

**Comment by nplanel** [comment]
run skydive-create-vagrant-boxes

**Comment by nplanel** [comment]
run skydive-create-vagrant-boxes

**Comment by nplanel** [comment]
run skydive-compile-tests

**Comment by nplanel** [comment]
run skydive-go-fmt

**Comment by nplanel** [comment]
run skydive-go-fmt

**Comment by nplanel** [comment]
run skydive-go-fmt

**Comment by nplanel** [comment]
run skydive-go-fmt

**Comment by nplanel** [comment]
run skydive-go-fmt

**Comment by nplanel** [comment]
run skydive-unit-tests

**Comment by nplanel** [comment]
run skydive-go-fmt

**Comment by nplanel** [comment]
run skydive-go-fmt

File: contrib/vagrant/setup-common.sh [hunk 6]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash
 
-sudo yum -y install python2 python-yaml
+sudo yum -y install python2 python-yaml libpcap
```
Comment (lebauce) [inline contrib/vagrant/setup-common.sh]: Why install libpcap here ?

**Pull request merged**